diff --git a/content/browser/renderer_host/render_widget_host_impl.cc b/content/browser/renderer_host/render_widget_host_impl.cc
index 60cfacdcf9fc..9b930b8acd1d 100644
--- a/content/browser/renderer_host/render_widget_host_impl.cc
+++ b/content/browser/renderer_host/render_widget_host_impl.cc
@@ -1028,6 +1028,11 @@ void RenderWidgetHostImpl::PauseForPendingResizeOrRepaints() {
   if (is_hidden())
     return;
 
+  // Do not pause if there is already a pending operation with the
+  // backing store.
+  if (in_get_backing_store_)
+    return;
+
   // Do not pause if there is not a paint or resize already coming.
   if (!visual_properties_ack_pending_)
     return;
