diff --git a/ui/latency/BUILD.gn b/ui/latency/BUILD.gn
index e9314a152362..fd30cc24af29 100644
--- a/ui/latency/BUILD.gn
+++ b/ui/latency/BUILD.gn
@@ -5,7 +5,7 @@
 import("//build/config/jumbo.gni")
 import("//testing/test.gni")
 
-jumbo_source_set("latency") {
+component("latency") {
   sources = [
     "fixed_point.cc",
     "fixed_point.h",
@@ -20,6 +20,7 @@ jumbo_source_set("latency") {
     "latency_tracker.h",
     "stream_analyzer.cc",
     "stream_analyzer.h",
+    "ui_latency_export.h",
     "windowed_analyzer.cc",
     "windowed_analyzer.h",
   ]
@@ -32,6 +33,8 @@ jumbo_source_set("latency") {
   public_deps = [
     "//services/metrics/public/cpp:metrics_cpp",
   ]
+
+  defines = [ "UI_LATENCY_IMPLEMENTATION" ]
 }
 
 jumbo_source_set("test_support") {
diff --git a/ui/latency/latency_info.h b/ui/latency/latency_info.h
index 43c61c8b9a4b..0acfdb344552 100644
--- a/ui/latency/latency_info.h
+++ b/ui/latency/latency_info.h
@@ -17,6 +17,7 @@
 #include "base/time/time.h"
 #include "services/metrics/public/cpp/ukm_source_id.h"
 #include "ui/gfx/geometry/point_f.h"
+#include "ui/latency/ui_latency_export.h"
 
 #if !defined(OS_IOS)
 #include "ipc/ipc_param_traits.h"  // nogncheck
@@ -123,7 +124,7 @@ enum SourceEventType {
   SOURCE_EVENT_TYPE_LAST = OTHER,
 };
 
-class LatencyInfo {
+class UI_LATENCY_EXPORT LatencyInfo {
  public:
   struct LatencyComponent {
     // Average time of events that happened in this component.
diff --git a/ui/latency/latency_tracker.h b/ui/latency/latency_tracker.h
index 4c62633f39ad..2513bac665f8 100644
--- a/ui/latency/latency_tracker.h
+++ b/ui/latency/latency_tracker.h
@@ -7,12 +7,13 @@
 
 #include "base/macros.h"
 #include "ui/latency/latency_info.h"
+#include "ui/latency/ui_latency_export.h"
 
 namespace ui {
 
 // Utility class for tracking the latency of events. Relies on LatencyInfo
 // components logged by content::RenderWidgetHostLatencyTracker.
-class LatencyTracker {
+class UI_LATENCY_EXPORT LatencyTracker {
  public:
   LatencyTracker();
   ~LatencyTracker() = default;
diff --git a/ui/latency/ui_latency_export.h b/ui/latency/ui_latency_export.h
new file mode 100644
index 000000000000..a29f5be94bcd
--- /dev/null
+++ b/ui/latency/ui_latency_export.h
@@ -0,0 +1,25 @@
+#ifndef UI_LATENCY_UI_LATENCY_EXPORT_H_
+#define UI_LATENCY_UI_LATENCY_EXPORT_H_
+
+#if defined(COMPONENT_BUILD)
+#if defined(WIN32)
+
+#if defined(UI_LATENCY_IMPLEMENTATION)
+#define UI_LATENCY_EXPORT __declspec(dllexport)
+#else
+#define UI_LATENCY_EXPORT __declspec(dllimport)
+#endif  // defined(UI_LATENCY_IMPLEMENTATION)
+
+#else  // defined(WIN32)
+#if defined(UI_LATENCY_IMPLEMENTATION)
+#define UI_LATENCY_EXPORT __attribute__((visibility("default")))
+#else
+#define UI_LATENCY_EXPORT
+#endif
+#endif
+
+#else  // defined(COMPONENT_BUILD)
+#define UI_LATENCY_EXPORT
+#endif
+
+#endif  // UI_LATENCY_UI_LATENCY_EXPORT_H_
