diff --git a/third_party/protobuf/BUILD.gn b/third_party/protobuf/BUILD.gn
index 3f8c1e6095eb..4d7ce806f4ce 100644
--- a/third_party/protobuf/BUILD.gn
+++ b/third_party/protobuf/BUILD.gn
@@ -623,6 +623,11 @@ if (current_toolchain == host_toolchain) {
 
     configs -= [ "//build/config/compiler:chromium_code" ]
     configs += [ "//build/config/compiler:no_chromium_code" ]
+    if (is_win) {
+      # https://crbug.com/703251
+      configs -= [ "//build/config/win:default_incremental_linking" ]
+      configs += [ "//build/config/win:no_incremental_linking" ]
+    }
 
     cflags = protobuf_lite_cflags
 
